Automattic/wp-calypso
# Export: 'Export Selected Items' functionality

~~Depends on #3359~~ Merged

This PR prepares a custom export request when the 'Export Selected Items' button is pressed in the Calypso exporter advanced settings.

#### ~~Todo~~
- [x] Store selected advanced settings in Redux store d859d31
- [x] Submit API request using selected settings fc77ca7
- [x] Fix existing unit tests (need to return a mock getState) 2ab03b4
- [x] Add unit tests for new redux actions & reducers 3ad4e43 13c59c3

### Testing instructions

1. Run the automated redux store tests
```sh
npm run test-client client/state/site-settings/exporter/test/*.js
```
2. Go to My Site > Settings > Export
3. Click the down chevron beside Export All
4. Select 'Posts' or 'Pages' and choose various options from the dropdowns
5. Open the Network requests tab in browser developer tools
6. Click 'Export Selected Items'
7. Find the request for `POST /sites/*.wordpress.com/exports/start`
8. Confirm that the form data request header matches the selected options. For example:

<img width="352" alt="screen shot 2016-05-19 at 6 10 12 pm" src="https://cloud.githubusercontent.com/assets/416133/15386783/ff29ea2c-1dec-11e6-9f8d-de631d7f74a5.png">

should match:

<img width="468" alt="screen shot 2016-05-19 at 6 10 02 pm" src="https://cloud.githubusercontent.com/assets/416133/15386794/055a1ce6-1ded-11e6-9cd7-26a3f6c27e63.png">

There are some server-side issues with the API endpoints which occasionally cause exports to fail, however these are not yet publicly in use and will be ironed out after this PR lands (which is still behind a feature flag)
